packages/babel-helper-create-class-features-plugin/src/index.ts

```diff
@@ -168,15 +168,18 @@ export function createClassFeaturePlugin({
 
         if (!props.length && !isDecorated) return;
 
-        let ref;
-
+        let ref = undefined;
         if (path.isClassExpression() || !path.node.id) {
           nameFunction(path);
           ref = path.scope.generateUidIdentifier("class");
         } else {
           ref = t.cloneNode(path.node.id);
         }
 
+        if (ref !== undefined) {
+          ref.innerBinding = path.node.id;
```

Comment:
Since `ref` is an AST node, we shouldn't use it to store additional information. It's better to keep `innetBinding` as a separate variable, and just do
```js
const innerBinding = path.node.id;
```

Also, we can move this to before the previous `if` (line 172): if a class doesn't have a `path.node.id` yet, it cannot be referenced anyway.